packages/oauth/oauth-provider/src/token/verify-token-claims.ts

```diff
@@ -25,77 +11,35 @@ export type VerifyTokenClaimsOptions = {
   scope?: [string, ...string[]]
 }
 
-export type VerifyTokenClaimsResult = {
-  token: OAuthAccessToken
-  tokenId: TokenId
-  tokenType: OAuthTokenType
-  tokenClaims: SignedTokenPayload
-  dpopProof: null | DpopProof
-}
-
 export function verifyTokenClaims(
-  token: OAuthAccessToken,
-  tokenId: TokenId,
   tokenType: OAuthTokenType,
   tokenClaims: SignedTokenPayload,
-  dpopProof: null | DpopProof,
   options?: VerifyTokenClaimsOptions,
-): VerifyTokenClaimsResult {
-  const dateReference = Date.now()
-
-  if (tokenClaims.cnf?.jkt) {
-    // An access token with a cnf.jkt claim must be a DPoP token
-    if (tokenType !== DPOP) {
-      throw new InvalidTokenError(
-        DPOP,
-        `Access token is bound to a DPoP proof, but token type is ${tokenType}`,
-      )
-    }
-
-    // DPoP token type must be used with a DPoP proof
-    if (!dpopProof) {
-      throw new InvalidDpopProofError(`DPoP proof required`)
-    }
-
-    // DPoP proof must be signed with the key that matches the "cnf" claim
-    if (tokenClaims.cnf.jkt !== dpopProof.jkt) {
-      throw new InvalidDpopKeyBindingError()
-    }
-  } else {
-    // An access token without a cnf.jkt claim must be a Bearer token
-    if (tokenType !== BEARER) {
-      throw new InvalidTokenError(
-        BEARER,
-        `Bearer token type must be used without a DPoP proof`,
-      )
-    }
-
-    // Unexpected DPoP proof received for a Bearer token
-    if (dpopProof) {
-      throw new InvalidTokenError(
-        BEARER,
-        `DPoP proof not expected for Bearer token type`,
-      )
-    }
-  }
-
+): void {
   if (options?.audience) {
-    const aud = asArray(tokenClaims.aud)
-    if (!options.audience.some((v) => aud.includes(v))) {
+    const { aud } = tokenClaims
+    const hasMatch =
+      aud != null &&
+      (Array.isArray(aud)
+        ? options.audience.some(includedIn, aud)
+        : options.audience.includes(aud))
+    if (!hasMatch) {
       throw new InvalidTokenError(tokenType, `Invalid audience`)
     }
   }
 
   if (options?.scope) {
     const scopes = tokenClaims.scope?.split(' ')
-    if (!scopes || !options.scope.some((v) => scopes.includes(v))) {
+    if (!scopes || !options.scope.some(includedIn, scopes)) {
```

Comment:
I think that the way this is written is not going to be familiar for many readers.  For that reason I'd recommend leaving it as it was but no biggie either way.